### ISSUE TITLE
Zoom event never raised

### ISSUE THREAD

**Issue by lucho78** [stephanrauh/ngx-extended-pdf-viewer#293]
Hi,
I've implemented the following code:
```html
<ngx-extended-pdf-viewer [hidden]="isLoading" [src]="src" height="100%" [(zoom)]="zoomValue"
                                 [useBrowserLocale]="true"
                                 [listenToURL]="false" [delayFirstView]="1000"
                                 [customToolbar]="additionalButtons"
                                 [(page)]="page"
                                 [language]="locale"
                                 (pdfLoaded)="onPDFLoaded($event)"
                                 (pagesLoaded)="onPageLoaded($event.pagesCount)"
                                 (pageChange)="onPageChange()" (zoomChange)="onZoomChange($event)">
```

But the event "zoomChange" is never raised.
Could you please check if it works for you in the version 3.3.1?
But the event currentZoomFactor seems to work as expected.

Thanks a lot
Luc

**Comment by stephanrauh** [comment]
Bonsoir Luc,

I've just scanned the issues. It seems the last time I've dealt with the `[zoom]` event, it was to solve a ticket of your's.

Maybe I've missed something, but chances are I didn't deliberately change anything involving the `zoom` property. Apart from #265. When I tried to reproduce your bug this evening, the zoom feature was entirely broken. Instead of zooming the PDF file the entire page was zoomed. That's exactly the opposite of what I tried to do with #265.

BTW, I'm pretty sure you can drop the property `delayFirstView="1000"`. It was necessary in the early days, before I learned how pdf.js initializes.

Best regards,
Stephan

**Comment by lucho78** [comment]
Hi Stephan,
Any news about this issue? I've tested with the latest version of the library but still the zoomChange is never raised.
Thanks
Luc

**Comment by stephanrauh** [comment]
I'm sorry I didn't find enough time to examine the issue. I hope I can do that this week-end.

**Comment by lucho78** [comment]
don't worry :-) and thanks a lot for your great support

**Comment by stephanrauh** [comment]
I believe I've found the bug. I also notice that Angular generates very strange JavaScript code. It seems I can simplify the code because I've simply repeated stuff Angular already does:
```javascript
PDFViewerApplication.eventBus.on('scalechanging', (/**
         * @param {?} x
         * @return {?}
         */
        (x) => {
            this.ngZone.run((/**
             * @return {?}
             */
            () => {
                setTimeout((/**
                 * @return {?}
                 */
                () => {
                    this.ngZone.run((/**
                     * @return {?}
                     */
                    () => {
                        this.currentZoomFactor.emit(x.scale);
                        if (this.zoom !== 'auto' && this.zoom !== 'page-fit' && this.zoom !== 'page-actual' && this.zoom !== 'page-width') {
                            this.emitZoomChange(x.scale * 100);
                        }
                    }));
                }));
            }));
        }));
```

**Comment by stephanrauh** [comment]
I've just published version 3.3.5. Please have a look if it's what you need. I'm afraid I've forgotten to define the precise meaning of `(zoomChange)` - it seems to be a contract between you, @lucho78, and me. :) Maybe you can also have a look at the updated documentation. It should arrive at https://pdfviewer.net/attributes in a few minutes. I've tried to define `(zoomChange)` more precisely.

**Comment by lucho78** [comment]
Hi Stephan,
Thanks a lot for your help on this topic
Basically I managed to do what I wanted. Now I get the value selected in the dropdown from pdfViewerApplication.pdfViewer.currentScaleValue
:-)
